src/BalanceTx/BalanceTx.purs

```diff
@@ -475,22 +476,22 @@ addTxCollateral utxos transaction =
 -- FIX ME: UnbalancedTx contains requiredSignatories which would be a part of
 -- multisig but we don't have such functionality ATM.
 
--- | Balances an unbalanced transaction. For submitting a tx via Nami, the
--- | utxo set shouldn't include the collateral which is vital for balancing.
--- | In particular, the transaction inputs must not include the collateral.
-balanceTx
-  :: UnattachedUnbalancedTx
+-- | Like `balanceTx`, but allows to provide an address that is treated like
+-- | user's own (while `balanceTx` gets it from the wallet).
+balanceTxWithOwnAddress
```

Comment:
I think the implementation is fine, but the name might be a bit confusing. `withOwnAddress` makes me think of the attached wallet's address, not an external user-provided address. Maybe just `withAddress` would be clearer?